********
Tutorial
********

This section provides a step-by-step tutorial for the :doc:`./case_study` and more.

.. note::
    
    Input data files to be imported into the MONGKIE in this tutorial were generated by the :download:`extractAlteredNet_GBM.R </assets/extractAlteredNet_GBM.R>`. Refer to :doc:`./case_study` for details about the method. Briefly, starting with somatic mutations, CNAs, and gene expression data sets downloaded from `UCSC Cancer Browser <https://genome-cancer.ucsc.edu/proj/site/hgHeatmap/#?bookmark=ce15f29a905207cbf3d0dbcdf9d35c18>`_, the :download:`R script </assets/extractAlteredNet_GBM.R>` does the following tasks.
    
    * Calculate expression log2FCs of each gene in normal vs. tumor conditions.
    * Calculate expression means of each gene in 4 GBM subtypes.
    * Extract a sub-network of altered (somatic mutation or CNV) genes in `STRING <http://string-db.org/>`_.
    * Calculate expression correlations between each pair of genes in the extracted network.
    * Write vertices, edges, and expression related data files.
    
    You can download the R script and generated input files in :download:`supplementary_data.zip </assets/supplementary_data.zip>`.

.. toctree::
    :caption: Table of Contents
    
    tutorial/import_network
    tutorial/visual_mapping
    tutorial/network_clustering
    tutorial/to_new_network
    tutorial/import_expression
    tutorial/functional_annotation
    tutorial/import_exp_profiles
    tutorial/export

